QUARKUS-2994.md

```diff
@@ -0,0 +1,43 @@
+# QUARKUS-2994 JDK 21 Support
+
+This document is a high level test plan for certification of Red Hat build of Quarkus on JDK 21.
+
+## Scope of the testing
+
+### Current test coverage
+* Regular CI runs in upstream include full testing for Java 21 (according to the announcement)
+* Daily runs on java 21 in GitHub Actions
+* Checks of code.quarkus in StartStop TS
+
+### Impact on test suites and testing automation
+* Temurin JVM 21 and OpenJDK 21 need to be added as JDKs to Quarkus QE Jenkins
+* Temurin JVM 21 and OpenJDK 21 have to be added as JVMs to Polarion
+* Bare metal jobs in extended platform testing need to be extended to run with both Temurin 21 and OpenJDK 21.
+  * This affects JVM mode only.
+* Openshift JVm job needs to use OpenJDK 21 (together with existing 17) instead of OpenJDK 11
+  * We also should use s2i build image based on jdk 21 there
+* Two quickstart jobs (Linux and Windows) should be added/repurposed from 11 in general or extended pipeline
```

Comment:
Imho, it would be useful to specify that quickstarts on RHEL + JDK21 one should be in general and on Win + JDK21 should be in extended as it's rather costly right now.

Reply:
Right now I am putting both to extended. Any reason, why RHEL version should be in general?

Reply:
For 3.2 we have mix of 11 and 17 for QS

Native job uses JDK 17, jvm one uses JDK 11.

3.8 should use JDK 21 for native and JDK 17 for jvm mode QS job.

Reply:
@rsvoboda our jobs in general pipeline are called
`rhbq-3.2-rhel8-jdk17-quickstarts-ts-native` and `rhbq-3.2-rhel8-jdk17-quickstarts-ts-jvm`. Did you mean some other trigger?